Sources/TokamakCore/Shapes/ShapeStyles/ContentStyles.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020-2021 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 7/7/21.
+//
+
+@frozen public struct PrimaryContentStyle {
+  @inlinable
+  public init() {}
+}
+
+extension PrimaryContentStyle: ShapeStyle {
+  public func _apply(to shape: inout _ShapeStyle_Shape) {
+    if !shape.inRecursiveStyle,
+       let foregroundStyle = shape.environment._foregroundStyle
+    {
+      if foregroundStyle.styles[0] is Self {
```

Comment:
Do we have a certainty that `foregroundStyle` is always non-empty? If so, maybe this needs a comment in code to explain that?

Also, this PR needs conflicts to be resolved 🙂

Reply:
I replaced the array with a tuple to guarantee its non-empty.